File: exercicios/Python/2152/2152.py
def main():
    n = int(input())
    h,m,o = 0,0,0
    for i in range(n):
        v = input().split()
        h = int(v[0])
        m = int(v[1])
        o = int(v[2])

        if o == 1:
            if h < 10 and m < 10:
                print('0{}:0{} - A porta abriu!'.format(h,m))
            
            elif h < 10 and m >= 10:
                print('0{}:{} - A porta abriu!'.format(h,m))
            
            elif h >= 10 and m < 10:
                print('{}:0{} - A porta abriu!'.format(h,m))
            
            elif h >= 10 and m >= 10:
                print('{}:{} - A porta abriu!'.format(h,m))
        
        if o == 0:
            if h < 10 and m < 10:
                print('0{}:0{} - A porta fechou!'.format(h,m))
            
            elif h < 10 and m >= 10:
                print('0{}:{} - A porta fechou!'.format(h,m))
            
            elif h >= 10 and m < 10:
                print('{}:0{} - A porta fechou!'.format(h,m))

            elif h >= 10 and m >= 10:
                print('{}:{} - A porta fechou!'.format(h,m))
    
main()

File: exercicios/JavaScript/2152/2152.js
var input = require('fs').readFileSync('stdin','utf8');
var lines = input.split('\n');

var x = parseInt(lines.shift());

for(let i=0; i<x; i++){

    var [h,m,o] = lines.shift().split(' ').map(Number)

    if(o == 1){
        if(h < 10 && m < 10){
            console.log(`0${h}:0${m} - A porta abriu!`)
        }else if(h > 9 && m < 10){
            console.log(`${h}:0${m} - A porta abriu!`)
        }else if(h < 10 && m > 9){
            console.log(`0${h}:${m} - A porta abriu!`)
        }else if(h > 9 && m > 9){
            console.log(`${h}:${m} - A porta abriu!`)
        }
    }else if(o == 0){
        if(h < 10 && m < 10){
            console.log(`0${h}:0${m} - A porta fechou!`)
        }else if(h > 9 && m < 10){
            console.log(`${h}:0${m} - A porta fechou!`)
        }else if(h < 10 && m > 9){
            console.log(`0${h}:${m} - A porta fechou!`)
        }else if(h > 9 && m > 9){
            console.log(`${h}:${m} - A porta fechou!`)
        }
    }
}